packages/@aws-cdk/core/README.md

```diff
@@ -305,6 +306,54 @@ new CustomResource(this, 'MyResource', {
 });
 ```
 
+#### The `core.CustomResourceProvider` class
+
+The class [`@aws-cdk/core.CustomResourceProvider`] offers a basic low-level
+framework designed to implement simple and slim custom resource providers. It
+currently only supports Node.js-based user handlers, and it does not have
+support for asynchronous waiting (handler cannot exceed the 15min lambda
+timeout).
+
+[`@aws-cdk/core.CustomResourceProvider`]: https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_core.CustomResourceProvider.html
+
+Set `serviceToken` to `provider.serviceToken` to use this provider:
+
+```ts
+import { CustomResource, CustomResourceProvider } from '@aws-cdk/core';
+const provider = new CustomResourceProvider(this, 'MyProvider', {
+  codeDirectory: `${__dirname}/my-handler`,
+  runtime: CustomResourceProviderRuntime.NODEJS_12, // currently the only supported runtime
+});
+new CustomResource(this, 'MyResource', {
+  serviceToken: provider.serviceToken
+});
+```
+
+The directory `my-handler` must include an `index.js` file which exports an async function named `handler`. This function accepts the CloudFormation resource event object and returns an object with the following structure:
+
+```js
+exports.handler = async function(event) {
+  const props = event.ResourceProperties;
+  const oldProps = event.OldResourceProperties; // in UPDATE
+  switch (event.RequestType) {
+    // ...
+  }
+  return {
+    // value of "Ref" (defaults to "event.PhysicalResourceId" or "event.RequestId")
```

Comment:
```suggestion
    // value of "REF" (defaults to "event.PhysicalResourceId" or "event.RequestId")
```

Reply:
Why? Isn't this `{ "Ref": "LogicalId" }` in cloudformation?

Reply:
OK, got it now. It was not clear with the `REF` just below. Maybe it's the `REF` then that is confusing...

Reply:
Fixed.